### ISSUE TITLE
categorical_feature support for LGBMClassifier in pipelines and *SearchCV estimators

### ISSUE THREAD

**Issue by RNarayan73** [microsoft/LightGBM#5594]
## Description
I am trying to use the categorical_feature in LGBMClassifier using the SKLearn API, by which I can avoid having to 1-hot encode the categorical features. The argument is available in the LGBMClassifier.fit() method as described here: [https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMClassifier.html#lightgbm.LGBMClassifier.fit]() which suggests that: 

- the default 'auto' value for categorical_feature will consider all pandas columns of the dtype 'category' as categorical features, 
- these columns need to be encoded as non-negative integers, as clarified here: [https://lightgbm.readthedocs.io/en/latest/Advanced-Topics.html#categorical-feature-support](url). I use the OrdinalEncoder to achieve this.

My LGBMClassifier instance is wrapped within a CalibratedClassifierCV (step 'clf') and occurs at the end of a set of transformations via the ColumnTransformer (step 'enc') and dimension reduction (step 'dim') and the whole pipeline is in turn wrapped for hyperparameter tuning within RandomizedSearchCV as in this diagram:
![image](https://user-images.githubusercontent.com/82275999/202875990-3eced61f-9a41-4140-88c6-031e495737ea.png)

So, after reading the input data via pd.read_csv(), I changed the desired columns grouped as 'target' and 'cat' above, to pandas dtype 'category' and fitted it to the estimator above.

I also tried a second scenario where I left the 'target' and 'cat' columns of the input df as the default dtypes as read by pd.read_csv and fitted it to the estimator.

I expected that in the first case (where I changed the pandas columns to dtype 'category'), the categories of the features would be used to split nodes and give me different results from the second case. However, in both instances, I got identical results suggesting that the features (with the same ordinal values in the columns in both cases) are not being treated any differently.

Am I missing something to get LightGBM to use the categorical features to split the nodes?

Note that the output of the ColumnTransformer step is a numpy array which gets propagated through the 'dim' step to the 'clf' steps. So, arguably, the data fed to LGBMClassifier at the end of the pipeline is not in a dataframe at all and the relevant columns won't therefore have the pandas dtype 'category' when fed to the .fit() method of LGBMClassifier.
Is this the reason for the identical results? If so, how can I utilise the categorical features functionality of LightGBM via the SKLearn API?

## Reproducible example
Hope above information is sufficient to answer my question. If you indeed need a more tangible estimator, please let me know.

## Environment info

LightGBM version or commit hash:
3.3.3

Command(s) you used to install LightGBM
```shell
pip install lightgbm
```

SKLearn: 1.1.2
Pandas: 1.5.1
numpy: 1.23.4

**Comment by jmoralez** [comment]
Hi @RNarayan73, thanks for using LightGBM. As you said the `auto` value checks for the categorical datatype if the input is a pandas dataframe, if it is a numpy array it won't be used. One possible solution I see (if I understand your problem correctly) is passing the feature indices to that argument instead, e.g. `clf__categorical_feature=[2, 5]` (assuming the 3rd and 6th columns that get to that step are the ones that you want to treat as categorical).

Please let us know if you have further doubts.
